ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -290,30 +249,56 @@ public boolean updateAccounts(Collection<Account> accounts) {
         accountServiceMetrics.notifyAccountDataChangeErrorCount.inc();
       }
     } else {
-      logger.error("Failed updating accounts={}", accounts);
+      logger.error("Failed updating accounts={}, took {} ms", accounts, timeForUpdate);
       accountServiceMetrics.updateAccountErrorCount.inc();
     }
     return hasSucceeded;
   }
 
   @Override
   public void close() {
-    if (isOpen) {
-      isOpen = false;
+    if (open.compareAndSet(true, false)) {
       if (scheduler != null) {
-        shutDownExecutorService(scheduler, storeConfig.accountUpdaterShutDownTimeoutMs, TimeUnit.MILLISECONDS);
+        shutDownExecutorService(scheduler, config.updaterShutDownTimeoutMs, TimeUnit.MILLISECONDS);
       }
       helixStore.stop();
     }
   }
 
+  /**
+   * To be used to subscribe to a {@link Notifier} topic. Upon receiving a
+   * {@link #FULL_ACCOUNT_METADATA_CHANGE_MESSAGE}, it will check for any account updates.
+   * @param topic The topic.
+   * @param message The message for the topic.
+   */
+  private void onAccountChangeMessage(String topic, String message) {
+    checkOpen();
+    logger.trace("Start to process message={} for topic={}", message, topic);
+    try {
+      switch (message) {
+        case FULL_ACCOUNT_METADATA_CHANGE_MESSAGE:
+          logger.trace("Start processing message={} for topic={}", message, topic);
+          readFullAccountAndUpdateCache(FULL_ACCOUNT_METADATA_PATH, true);
+          logger.trace("Completed processing message={} for topic={}", message, topic);
+          break;
+        default:
+          accountServiceMetrics.unrecognizedMessageErrorCount.inc();
+          throw new RuntimeException("Could not understand message=" + message + " for topic=" + topic);
+      }
+    } catch (Exception e) {
+      logger.error("Exception occurred when processing message={} for topic={}.", message, topic, e);
+      accountServiceMetrics.remoteDataCorruptionErrorCount.inc();
+      accountServiceMetrics.fetchRemoteAccountErrorCount.inc();
+    }
+  }
+
   /**
    * Reads the full set of {@link Account} metadata from {@link HelixPropertyStore}, and update the local cache.
    *
    * @param pathToFullAccountMetadata The path to read the full set of {@link Account} metadata.
-   * @param isCalledFromListener {@code true} if the caller is the account update listener, {@@code false} otherwise.
```

Comment:
I like the convention of prefixing booleans with 'is' myself, but I'm fine either way.  I'm guessing if(calledFromListener) looks nicer than if(isCalledFromListener)

Reply:
I don't have a strong preference, but I tend to not include the "is" in booleans. I just changed this one back to the way it was. Other folks have probably followed the "is" convention too.